src/Images.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Images
+  ( createImageTable,
+    getImageFromDisk,
+    storeImageInDB,
+    handleGetImage,
+    handleGetAssociatedImageUUIDs,
+  )
+where
+
+import Codec.Picture (DynamicImage, readImage)
+import Codec.Picture.Saving (imageToJpg)
+import Control.Monad.IO.Class (MonadIO (liftIO))
+import Data.ByteString.Lazy qualified as BS (writeFile)
+import Data.UUID (UUID)
+import Data.UUID.V4 (nextRandom)
+import Database.SQLite.Simple (Connection, FromRow (fromRow), Only (Only), ToRow (toRow), execute, execute_, field, query)
+import Database.SQLite.Simple.ToField (ToField (toField))
+import Network.HTTP.Types (status400, status404)
+import Recipe (RecipeID)
+import System.FilePath ((<.>), (</>))
+import Text.Read (readMaybe)
+import Web.Scotty (ActionM, addHeader, json, param, raiseStatus, raw)
+
+createImageTable :: Connection -> IO ()
+createImageTable db = do
+  {-
+    Image:
+      uuid:     UUID of the image
+      title:    Title of the image
+      filePath: Filepath to the image
+      recipeID: ID of the associated recipe
+  -}
+  execute_
+    db
+    "CREATE TABLE IF NOT EXISTS images (\
+    \ uuid VARCHAR(36) PRIMARY KEY NOT NULL,\
+    \ title VARCHAR(40) NOT NULL,\
+    \ recipeId INTEGER,\
+    \ FOREIGN KEY (recipeId) REFERENCES recipes(recipeId));"
+
+data ImageInfo = ImageInfo
+  { uuid :: UUID,
+    title :: String,
+    recipeId :: RecipeID
+  }
+
+instance FromRow ImageInfo where
+  fromRow = ImageInfo <$> (read <$> field) <*> field <*> field
+
+instance ToRow ImageInfo where
+  toRow (ImageInfo imageUUID imageTitle rId) = [toField . show $ imageUUID, toField imageTitle, toField rId]
+
+getImageFromDisk :: FilePath -> UUID -> IO (Maybe DynamicImage)
+getImageFromDisk imageFolder imageUUID = either (const Nothing) Just <$> (readImage . imagePath imageFolder $ imageUUID)
+
+storeImageOnDisk :: FilePath -> DynamicImage -> IO ()
+storeImageOnDisk fPath = BS.writeFile fPath . imageToJpg 50
```

Comment:
Wenn 50 die Quality ist, muss das mind. 80 sein. 50 ist viel zu wenig. 80-90

Reply:
Quality changed to 80 by d1ab51931033a71d4ad34948b1ba13ad35e84793 (Maybe add this to the config in the future?)